### ISSUE TITLE
Multi-stage Kmeans performance comparison with Petuum

### ISSUE THREAD

**Issue by liuzhanhao** [Yuzhen11/flexps-paper-version#66]
# Multi-stage Kmeans performance on PowerPS

## Configuration & Result:
`Data set: ` [mni8m](https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/multiclass.html#mnist8m)
\# of classes: 10
\# of data: 8,100,000
\# of features: 784
\# of loading workers: 10

### Multi-stage with different \# of workers
| Stage        | # of worker          | # of iterations  | batch size | learning rate coefficient | time used
| ------------- |:-------------:| -----:|-----:|-----:|-----:|
| Stage 1     | 5 |  10000| 100 | 0.03 | 34.114s |
| Stage 2      | 10      |  10000| 300| 0.03 | 47.240s | 
| Stage 3  | 20     |    10000| 1000 |0.03 | 75.341s |

`Total time: 218.339s`  (including loading task, init task and training task)

### Multi-stage with constant \# of workers
| Stage        | # of worker          | # of iterations  | batch size | learning rate coefficient | time used
| ------------- |:-------------:| -----:|-----:|-----:|-----:|
| Stage 1     | 20 |  10000| 100 | 0.03 | 73.993s |
| Stage 2      | 20      |  10000| 300| 0.03 | 74.483s | 
| Stage 3  | 20     |    10000| 1000 |0.03 | 75.264s |

`Total time: 292.449s`  (including loading task, init task and training task)
